src/entities/asset/AssetIcon.vue

```diff
@@ -0,0 +1,69 @@
+<template>
+  <div class="asset">
+    <div class="asset__icon">
+      <XOR v-if="nameLower=='xor'" />
+      <VAL v-else-if="nameLower=='val'" />
+      <ETH v-else-if="nameLower=='eth'" />
+      <IrohaIcon v-else-if="nameLower=='iroha'" />
+      <Soshiba v-else-if="nameLower=='soshiba'" />
+      <Pswap v-else-if="nameLower=='pswap'" />
+      <NaIcon v-else />
```

Comment:
Written in JS, it could be less verbose:

```ts
const ICONS = {
  xor: XOR,
  // ...
}

const Icon = computed(() => ICONS[nameLower.value] ?? NaIcon)
```

```vue
<template>
  <component :is="Icon" />
</template>
```